.github/bin/lint-whitespace

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env sh
+
+# Change directory to the repo root, so that `git grep`:
+# - Operates on the whole repo, even if this script is called from a subdirectory.
+# - Outputs paths relative to the repo root.
+repo_root="$(git rev-parse --show-toplevel)"
+cd "${repo_root}" || exit
+exit_status=0
+
+# Check that each git-tracked non-binary file has no trailing whitespace.
+trailing_whitespace="$(git grep --cached -I '[[:blank:]]$')"
```

Comment:
This repo doesn't have any binary files, but it's generally important to include the `-I` option - otherwise the script can exit with an error for "trailing whitespace" in a binary file.

There is at least one action in the GitHub Marketplace that doesn't handle that correctly.

Docs for `git grep`: https://git-scm.com/docs/git-grep